packages/lodestar/src/chain/validation/aggregateAndProof.ts

```diff
@@ -24,126 +22,124 @@ export async function validateGossipAggregateAndProof(
-
   if (attestationSlot < earliestPermissibleSlot) {
-    throw new AttestationError({
+    throw new AttestationGossipError(GossipAction.IGNORE, {
       code: AttestationErrorCode.PAST_SLOT,
       earliestPermissibleSlot,
       attestationSlot,
       job: attestationJob,
     });
   }
-
   if (attestationSlot > latestPermissibleSlot) {
-    throw new AttestationError({
+    throw new AttestationGossipError(GossipAction.IGNORE, {
       code: AttestationErrorCode.FUTURE_SLOT,
       attestationSlot,
       latestPermissibleSlot,
       job: attestationJob,
     });
   }
 
+  // [REJECT] The aggregate attestation's epoch matches its target
+  // -- i.e. aggregate.data.target.epoch == compute_epoch_at_slot(aggregate.data.slot)
+
+  // [IGNORE] The aggregate is the first valid aggregate received for the aggregator with
+  // index aggregate_and_proof.aggregator_index for the epoch aggregate.data.target.epoch.
   if (db.seenAttestationCache.hasAggregateAndProof(aggregateAndProof)) {
-    throw new AttestationError({
+    throw new AttestationGossipError(GossipAction.REJECT, {
       code: AttestationErrorCode.AGGREGATE_ALREADY_KNOWN,
       job: attestationJob,
     });
   }
 
-  if (!hasAttestationParticipants(aggregate)) {
+  // [REJECT] The attestation has participants -- that is,
+  // len(get_attesting_indices(state, aggregate.data, aggregate.aggregation_bits)) >= 1.
+  if (indexedAttestation.attestingIndices.length < 1) {
     // missing attestation participants
-    throw new AttestationError({
+    throw new AttestationGossipError(GossipAction.REJECT, {
       code: AttestationErrorCode.WRONG_NUMBER_OF_AGGREGATION_BITS,
       job: attestationJob,
     });
   }
 
-  if (await isAttestingToInValidBlock(db, aggregate)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.KNOWN_BAD_BLOCK,
-      job: attestationJob,
-    });
-  }
-
-  await validateAggregateAttestation(config, chain, signedAggregateAndProof, attestationJob);
-}
-
-export function hasAttestationParticipants(attestation: phase0.Attestation): boolean {
-  return Array.from(attestation.aggregationBits).filter((bit) => !!bit).length >= 1;
-}
-
-export async function validateAggregateAttestation(
-  config: IBeaconConfig,
-  chain: IBeaconChain,
-  aggregateAndProof: phase0.SignedAggregateAndProof,
-  attestationJob: IAttestationJob
-): Promise<void> {
-  const attestation = aggregateAndProof.message.aggregate;
-
-  let attestationTargetState: CachedBeaconState<allForks.BeaconState>;
+  let committee: ValidatorIndex[];
   try {
-    // the target state, advanced to the attestation slot
-    attestationTargetState = await chain.regen.getCheckpointState(attestation.data.target);
-  } catch (e) {
-    throw new AttestationError({
-      code: AttestationErrorCode.MISSING_ATTESTATION_TARGET_STATE,
-      error: e as Error,
+    committee = attestationTargetState.getBeaconCommittee(aggregate.data.slot, aggregate.data.index);
```

Comment:
not important but the `aggregateAndProof` is a TreeBacked here so we want to get each property (for example `aggregate.data`) only one time